problem_builder/mixins.py

```diff
@@ -176,3 +177,25 @@ class NoSettingsMixin(object):
     def studio_view(self, _context=None):
         """ Studio View """
         return Fragment(u'<p>{}</p>'.format(self._("This XBlock does not have any settings.")))
+
+
+class StudentViewUserStateMixin(object):
+    NESTED_BLOCKS_KEY = "components"
+    INCLUDE_SCOPES = (Scope.user_state, Scope.user_info, Scope.preferences)
+
+    def student_user_view_user_state(self, context=None):
```

Comment:
In general, `student_user_view_user_state` may not simply be a set of fields... we may want to simplify/transform the data first. I think it's fine in this case though.